Slim/Schema.pm

```diff
@@ -2359,13 +2359,8 @@ sub artistOnlyRoles {
 		}
 	}
 
-	# If we're using all roles, don't bother with the constraint.
-	if (scalar keys %roles != Slim::Schema::Contributor->totalContributorRoles) {
-
-		return [ sort map { Slim::Schema::Contributor->typeToRole($_) } keys %roles ];
-	}
+	return [ sort map { Slim::Schema::Contributor->typeToRole($_) } keys %roles ];
 
-	return undef;
```

Comment:
I'm a bit lost here... why would you remove the condition?

Reply:
That was the whole point: if all roles were selected, the condition is false, so it dropped through to `return undef; `which causes the problem I mentioned in `artistsQuery`. Previously, because we never included `TRACKARTIST`, the condition was always true.